src/hoc/withHfpData.js

```diff
@@ -26,10 +26,12 @@ const getGroupedByVehicle = (hfpData) => {
   }
 
   const groupedData = groupBy(hfpData, "uniqueVehicleId");
-  return map(groupedData, (positions, groupName) => ({
+  const vehicleGroups = map(groupedData, (positions, groupName) => ({
     vehicleId: groupName,
     positions,
   }));
+
```

Comment:
@varpushaukka, add console.time() to the start and end of these functions to measure how they perform on your machine.